### DISCUSSION TITLE
Is there any way to use Tensorboard on the Planetary Computer Hub?

### DISCUSSION THREAD

**Discussion by valentina-s** [microsoft/PlanetaryComputer#38] [Ideas]
Locally, one run it on a separate port, but I am not sure if one can access other ports through the hub. On Colab, one can use [ngrok](https://ngrok.com/).

**Comment by TomAugspurger** [comment]
Once you start the tensorboard server (`tensorboard --logdir=mylogdir`) You should be able to the UI at `{hub-url}/user/{username}/proxy/{port}/`. Since my username is `taugspurger@microsoft.com` that would be https://pccompute.westeurope.cloudapp.azure.com/compute/user/taugspurger@microsoft.com/proxy/6006/. **Note the trailing `/` is important**.

<img width="946" alt="image" src="https://user-images.githubusercontent.com/1312546/157915013-0db56a7e-f922-4034-8206-6f9f516fbe84.png">.

**Comment by valentina-s** [comment]
Thanks @TomAugspurger for suggesting to use Jupyter Server Proxy! I tested it with this [tutorial notebook](https://www.tensorflow.org/tensorboard/get_started).  Once I run the command from the hub terminal
`tensorboard --logdir logs/fit --bind_all`
I go to https://pccompute.westeurope.cloudapp.azure.com/compute/user/vms16@uw.edu/proxy/6006
and get a blank page but with Tensorboard title so seems the port is talking to Tensorboard but cannot display anything. The error in the Chrome is inspector is 
 https://pccompute.westeurope.cloudapp.azure.com/compute/user/vms16@uw.edu/proxy/index.js?_file_hash=29a7d03a net::ERR_ABORTED 404

I tested tensorboard 2.8.0 and tensorboard 2.7.0. 

Any suggestions? Am I using it right?

**Reply by TomAugspurger** [reply]
Can you check whether adding a trailing slash fixes things for you? https://github.com/tensorflow/tensorboard/issues/1270#issuecomment-402273281 indicates that might work.

I'm a bit pessimistic since you say some things like the title are working, but it's worth a shot.

When working on that other example using tensorboard, I did notice that the tensorboard javascript took a long time to load (maybe >60s?) But it didn't throw a 404, it did eventually load.

**Reply by TomAugspurger** [reply]
I confirmed that opening a url to `/proxy/6006` failed, but `/proxy/6006/` did seem to work. I'll update my original answer.

<img width="946" alt="image" src="https://user-images.githubusercontent.com/1312546/157914865-b2c87654-c230-4946-98a4-06dd2f04ebda.png">

**Reply by valentina-s** [reply]
Thanks, I got it working (sorry for the delay, had the project on pause while my colleague was on a field trip). 

I noticed a strange behavior in the hub terminal though: when I write a command with a path and use up arrow to edit it a previous run, it will look like I have run the new path but in fact the old path was run (I could see the old path in tensorboard). I had to type the command explicitly to resolve it. Seems it is some update issue while typing in the terminal in the browser (I know this off-topic but thought to mention it since it definitely confused me for a while).
